community/kernel/src/main/java/org/neo4j/kernel/impl/store/kvstore/KeyValueStoreFile.java

```diff
@@ -0,0 +1,326 @@
+/**
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.store.kvstore;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Arrays;
+
+import org.neo4j.io.pagecache.PageCursor;
+import org.neo4j.io.pagecache.PagedFile;
+
+import static org.neo4j.io.pagecache.PagedFile.PF_NO_GROW;
+import static org.neo4j.io.pagecache.PagedFile.PF_SHARED_LOCK;
+import static org.neo4j.kernel.impl.store.kvstore.BigEndianByteArrayBuffer.buffer;
+import static org.neo4j.kernel.impl.store.kvstore.BigEndianByteArrayBuffer.compare;
+
+/**
+ * Stores Key/Value pairs sorted by the key in unsigned big-endian order.
+ *
+ * @param <META> the type of object to hold the metadata of a store file.
+ */
+public class KeyValueStoreFile<META> implements Closeable
+{
+    public META metadata()
+    {
+        return metadata;
+    }
+
+    /**
+     * Visit key value pairs that are greater than or equal to the specified key. Visitation will continue as long as
+     * the visitor {@link KeyValueVisitor#visit(ReadableBuffer, ReadableBuffer) returns true}. The only situation
+     *
+     * @return {@code true} if an exact match was found, meaning that the first visited key/value pair was a perfect
+     * match for the specified key.
+     */
+    public boolean scan( SearchKey the, KeyValueVisitor visitor ) throws IOException
```

Comment:
dammit, another one of those cases of being interrupted while writing docs, I'll rewrite all of this doc comment, I don't remember what I was thinking at the time.